src/include/optimizer/rules/unnesting_rules.h

```diff
@@ -120,4 +123,20 @@ class DependentSingleJoinToInnerJoin : public Rule {
                  OptimizationContext *context) const override;
 };
 
+/**
+ * Given predicates associated with some aggregate, this function will extract the predicates that are correlated with
+ * an outer query, as well as the columns from those predicates that aren't correlated.
+ * @param predicates vector of predicates associated with an aggregate
+ * @param child_group_aliases_set the table alias set of the predicate's child node
+ * @return
+ *      correlated_predicates - predicates which are correlated to an outer query
+ *      normal_predicates - predicates which are not correlated to an outer query
+ *      new_group_cols - columns from correlated predicates which are not correlated to an outer query. These will be
+ * used as group by columns in the nested aggregate
+ */
+std::tuple<std::vector<AnnotatedExpression>, std::vector<AnnotatedExpression>,
```

Comment:
Is it not good practice to use `tuple`s like this?

Reply:
Would it be better to just make the input params to avoid moving the results?

Reply:
I don't know if we have a codified good practice or not around `tuple`.
IIRC there's something about `tuple` potentially producing stupider code on gcc compared to clang.
I don't know enough C++ to have a good opinion here.

Perhaps input params are better in favor of readability though.

Reply:
I switched to the pointer approach